### PR TITLE
docs: add docs/ folder and enhance review checklist

### PR DESCRIPTION
**Pull request by amitjoshi-ms**
## Summary

This PR enhances project documentation structure and improves the code review process.

### Project Structure Updates
- Added `docs/` folder for user-facing documentation
- Updated `copilot-instructions.md` with clear folder descriptions:
  - `src/` - Application source code
  - `public/` - Static assets
  - `tests/` - Unit and E2E tests
  - `docs/` - User-facing documentation
  - `specs/` - Spec-kit feature specifications (auto-generated)

### Review Checklist Enhancements
Added two new sections to `review.task.prompt.md`:

**Testing Gates**
- New functionality has corresponding unit tests
- Edge cases and error conditions are tested
- Test assertions are meaningful
- Tests are independent and don't rely on execution order

**AI Tooling & Instructions**
- Check if `copilot-instructions.md` needs updates
- Check if prompts in `.github/prompts/` need updates
- Check if agents in `.github/agents/` need updates
- Check if documentation in `docs/` needs updates

## Testing
- ✅ All 77 unit tests pass
- ✅ Linting passes

### CODE REVIEW

File: .github/prompts/review.task.prompt.md [hunk 1]
```diff
@@ -108,6 +114,12 @@ Perform a comprehensive review of current code changes covering code quality, se
 - [ ] Complex logic has explanatory comments
 - [ ] README updated if needed
 
+### AI Tooling & Instructions
+- [ ] `copilot-instructions.md` updated if project structure, commands, or conventions changed
+- [ ] Relevant prompts in `.github/prompts/` updated if workflows or processes changed
+- [ ] Relevant agents in `.github/agents/` updated if agent behaviors need adjustment
+- [ ] Documentation in `docs/` updated if user-facing features or APIs changed
```
[the remark this answers is not in the document]
Comment (amitjoshi-ms) [reply .github/prompts/review.task.prompt.md]: keep it.
